ocaml/rrdd/rrdd_main.ml

```diff
@@ -153,9 +153,13 @@ module Meminfo = struct
 				let meminfo_free = Int64.of_string (xs.Xs.read path) in
 				debug "memfree has changed to %Ld in domain %d" meminfo_free d;
 				current_meminfofree_values := IntMap.add d meminfo_free !current_meminfofree_values
-			with Xenbus.Xb.Noent ->
+			with
+			  Xenbus.Xb.Noent ->
 				debug "Couldn't read path %s; forgetting last known memfree value for domain %d" path d;
 				current_meminfofree_values := IntMap.remove d !current_meminfofree_values
+			| Failure("int_of_string") ->
```

Comment:
My thought was that if anything other than "unable to read" or "unable to parse" happens, something very very unusual have to happen and it might be a good idea to fail miserably and be informed of the issue.
But thinking again about it, it might make more sense to completely avoid any disruption as you suggest.